### PR TITLE
Add a new ImageEmbeddingTable class

### PR DESCRIPTION
**Pull request by sasxianhu**
Provide several APIs to load and show the data for image embedding.

### CODE REVIEW

File: dlpy/image_embedding.py [hunk 1]
```diff
@@ -0,0 +1,701 @@
...
+import os
+
+from swat.cas.table import CASTable
+
+from dlpy import ImageTable
+from .utils import random_name, image_blocksize, caslibify_context, get_server_path_sep, get_cas_host_type
+from warnings import warn
+
+
+class ImageEmbeddingTable(ImageTable):
```
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: Not now. In theory, we can do it for text.

File: dlpy/image_embedding.py [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+
+        Create ImageEmbeddingTable from files in `path`
+
+        Parameters
+        ----------
+        conn : CAS
+            The CAS connection object
+        path : string
+            The path to the image directory on the server.
+            Path may be absolute, or relative to caslib root if specified.
```
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: I prefer to leave it as it is for now since Kyle is working on docs now. This description is same as the one from images.py now.

File: dlpy/image_embedding.py [hunk 3]
```diff
@@ -0,0 +1,701 @@
...
+
+        conn.loadactionset('image', _messagelevel='error')
+        conn.loadactionset('sampling', _messagelevel='error')
+        conn.loadactionset('deepLearn', _messagelevel='error')
+
+        if casout is None:
+            casout = dict(name=random_name())
+        elif isinstance(casout, CASTable):
+            casout = casout.to_outtable_params()
+
```
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: It is same as images.py and other files also use the similar logic. It does not take a string.

File: dlpy/image_embedding.py [hunk 4]
```diff
@@ -0,0 +1,701 @@
...
+        # remove the csv file
+        conn.deleteSource(source=file_list_casout['name'] + '.csv', caslib=caslib, _messagelevel='error')
+
+        # reset msg level
+        conn.setSessOpt(messageLevel=current_msg_level['messageLevel'])
+
+        return images_casout
+
+    @staticmethod
```
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: I use static to indicate the method is only for internal usage.

File: dlpy/image_embedding.py [hunk 5]
```diff
@@ -0,0 +1,701 @@
...
+          for Siamese, frequency for the first + second image label columns and
+          frequency for the dissimilar column
+          for triplet and quartet, frequency for the third image label column,
+          for triplet, frequency for the first + second + third image label columns
+          for quartet, frequency for the fourth image label column,
+          for quartet, frequency for the first + second + third + fourth image label columns
+        )
+        '''
+
+        if self.embedding_model_type.lower() == 'siamese':
```
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: I tried before. Then I found out the current way is much simpler without duplicating too many codes.

File: dlpy/tests/test_image_embedding_table.py [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+            cls.data_dir = os.environ.get('DLPY_DATA_DIR')
+            if cls.data_dir.endswith(cls.server_sep):
+                cls.data_dir = cls.data_dir[:-1]
+            cls.data_dir += cls.server_sep
+
+        if "DLPY_DATA_DIR_LOCAL" in os.environ:
+            cls.local_dir = os.environ.get("DLPY_DATA_DIR_LOCAL")
+
+        # the server path that points to DLPY_DATA_DIR_LOCAL
+        if "DLPY_DATA_DIR_SERVER" in os.environ:
```
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/tests/test_image_embedding_table.py]: Not sure what you mean.

File: dlpy/image_embedding.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/image_embedding.py]: we also need to add this to images.py as well then.

File: dlpy/image_embedding.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/image_embedding.py]: __ already makes it private.

File: dlpy/image_embedding.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: Not just images.py. Many other py files use the same way to accept casout.

File: dlpy/image_embedding.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/image_embedding.py]: this way is confusing and it is hard to follow.

File: dlpy/tests/test_image_embedding_table.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/tests/test_image_embedding_table.py]: something like checking if the server path ends with a server_sep do nothing, otherwise add one so that we dont mind if the user pass a path with an ending slash or not.

File: dlpy/image_embedding.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: https://www.geeksforgeeks.org/class-method-vs-static-method-python/
A static method does not receive an implicit first argument.
This is what I checked.

File: dlpy/image_embedding.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/image_embedding.py]: I will open a story for this. we can change all off them to be consistent.

[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/image_embedding.py]: thanks for pointing out

File: dlpy/image_embedding.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/image_embedding.py]: yes but it does not make it private. if you want a function for only internal use. then you make it private. static make it callable without a self argument. (e.g., Object.function() ) Depends on what you want. if you want to have your function to be called like (Object.function()) keep it.

[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/image_embedding.py]: https://medium.com/@manjuladube/encapsulation-abstraction-35999b0a3911

File: dlpy/image_embedding.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: I know. But I decided to use the current style. I indeed tried functions before as I mentioned. Still ugly as well. So I ended up using the current way.

File: dlpy/image_embedding.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: Yes, do not use an implicit first argument.

[the remark this answers is not in the document]
Comment (ethem-kinginthenorth) [reply dlpy/image_embedding.py]: if you are not going to call this function out of the class, I still dont see any point of having  this function as static.

File: dlpy/tests/test_image_embedding_table.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/tests/test_image_embedding_table.py]: I see. I added the check to server_dir but not local_dir.
I actually do not use these envs for this py file but just added them for future uses.

File: dlpy/image_embedding.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (sasxianhu) [reply dlpy/image_embedding.py]: I know why I use static here: make Pycharm happy. If I remove it, it keeps showing me the first argument should be self. Really annoying. So I added static.
